daceml/onnx/op_implementations/fpga_implementations.py

```diff
@@ -2807,17 +2799,187 @@ def forward(node: ONNXOp, state: SDFGState,
                                   memlet=dace.Memlet("sum_res[0]"))
         new_state.add_memlet_path(outer_me, init_tasklet, memlet=dace.Memlet())
 
-
         new_state.add_memlet_path(store_tasklet,
                                   outer_mx,
                                   out_data,
                                   src_conn="out_res",
                                   memlet=dace.Memlet("reduced[o0, o1, o2]"))
 
-
-
-
         new_sdfg.fill_scope_connectors()
         new_sdfg.validate()
         new_sdfg.save('/tmp/reduce_sum.sdfg')
         return new_sdfg
+
+
+# -----------------------------
+# Binary element-wise operators
+# -----------------------------
+def binary_forward(node: ONNXOp, state: SDFGState, sdfg: SDFG,
+                   operator: str) -> typing.Union[Node, SDFG]:
+
+    node.validate(sdfg, state)
+
+    op_map = {'+': 'add', '-': 'sub', '*': 'mul', '/': 'div'}
+    op_name = op_map[operator]
+
+    streaming_node = False
+
+    # Get inputs and outputs names
+    A = in_desc_with_name(node, state, sdfg, "A")
+    B = in_desc_with_name(node, state, sdfg, "B")
+    C = out_desc_with_name(node, state, sdfg, "C")
+
+    # Create inner SDFG
+    op_sdfg = dace.SDFG(node.name)
+    compute_state = op_sdfg.add_state("compute_{}".format(op_name))
+
+    a_inner = A.clone()
+    a_inner.transient = False
+    b_inner = B.clone()
+    b_inner.transient = False
+    c_inner = C.clone()
+    c_inner.transient = False
+
+    op_sdfg.add_datadesc("A", a_inner)
+    op_sdfg.add_datadesc("B", b_inner)
+    op_sdfg.add_datadesc("C", c_inner)
+
+    # Create map and tasklet
+    map_ranges = {'__i%d' % i: '0:%s' % n for i, n in enumerate(A.shape)}
```

Comment:
You have a mix between two kinds of string formatting:
- `'__i%d' % i`
- `'__i{}'.format(i)`

We have actually started to prefer the newest one:
- `f'{i}'` (called f-strings, see [here](https://docs.python.org/3/tutorial/inputoutput.html#tut-f-strings))

Try to get used to using f-strings!

Also: Be consistent with using single or double quotes within a single file (we are not consistent across DaCe, but at least within a file/function it would be good to stick to one).